src/test/java/seedu/address/model/person/PersonTest.java

```diff
@@ -32,9 +32,14 @@ public void isSamePerson() {
         // null -> returns false
         assertFalse(ALICE.isSamePerson(null));
 
-        // same name, all other attributes different -> returns true
+        // same name, all other attributes different -> returns false
         Person editedAlice = new PersonBuilder(ALICE).withPhone(VALID_PHONE_BOB).withEmail(VALID_EMAIL_BOB)
                 .withAddress(VALID_ADDRESS_BOB).withTags(VALID_TAG_HUSBAND).build();
+        assertFalse(ALICE.isSamePerson(editedAlice));
+
+        // same name, same phone number, all other attributes different -> returns true
+        editedAlice = new PersonBuilder(ALICE).withEmail(VALID_EMAIL_BOB).withAddress(VALID_ADDRESS_BOB)
+                .withTags(VALID_TAG_HUSBAND).build();
```

Comment:
nit: reuse of variable but for different logic. I think renaming to `aliceSameName` and `aliceSameNameAndNumber` might be better for clarity and maintainability.

Reply:
Good catch, I've edited this accordingly!